KAS-4496: reordering agendaitems now doesnt automatically save after each move + added sum helper

StuyckRuben opened the pull request:
https://kanselarij.atlassian.net/browse/KAS-4496

Commit:
KAS-4496: reordering agendaitems now doesnt automatically save after each move + added sum helper

sergiofenoll committed:
Merge branch 'development' into feature/KAS-4496-efficientie-volgorde-agendapunten-aanpassen-verbeteren

sergiofenoll commented on app/templates/agenda/agendaitems.hbs:
```diff
@@ -38,7 +38,11 @@
         @searchText={{readonly this.filter}}
         @onSearch={{this.searchAgendaitems}}
-      />
+        @showReorderButton="true"
```
```suggestion
        @showReorderButton={{not this.showModifiedOnly}}
```

When we show modified only we don't allow re-ordering (so e.g. we hide the `ChangesAlert` component), if you still click this button while in this state you get stuck: the button becomes greyed out and the alert that has buttons to cancel is not shown.

Better to just hide this button when in a state we don't support.

sergiofenoll commented on app/components/agenda/agenda-overview.hbs:
```diff
@@ -1,9 +1,11 @@
 <div class="vlc-agenda-items" ...attributes>
-  {{#if @isEditingOverview}}
+  {{#if (and @isEditingOverview this.canMoveAgendaitems)}}
```
```suggestion
  {{#if this.canMoveAgendaitems}}
```
`this.canMoveAgendaitems` already contains `this.args.isEditingOverview`

sergiofenoll commented on app/components/agenda/agenda-overview.hbs:
```diff
@@ -105,9 +125,9 @@
+    {{#if this.announcements.length}}
       <ul {{sortable-group onChange=@onReorderAgendaitems}}>
-        {{#each @announcements as |announcement indexitems|}}
+        {{#each this.announcements as |announcement indexitems|}}
```
```suggestion
        {{#each this.announcements as |announcement indexItems|}}
```
Casing here is different, better to match `camelCasing`

sergiofenoll commented on app/components/agenda/agenda-overview.hbs:
```diff
@@ -127,7 +147,8 @@
               @allowReorderItems={{this.canMoveAgendaitems}}
               @onMove={{this.move}}
               @isFirst={{eq indexitems 0}}
```
```suggestion
              @isFirst={{eq indexItems 0}}
```

sergiofenoll commented on app/components/agenda/agenda-overview.hbs:
There's a longstanding bug here where when re-ordering in an agenda with notas and announcements, the two lists will get merged. Before it didn't matter because as soon as a user stopped dragging we would save the changes and reset the view, but now it's important to get this working.

Screencast:
[Screencast from 2024-02-12 10-18-52.webm](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/assets/22411874/98a3d14f-e176-4ddb-8ed1-d760a3974923)

This seems to be a known thing with the addon we use, see [this part of their README](https://github.com/adopted-ember-addons/ember-sortable/tree/v4.0.3?tab=readme-ov-file#multiple-ember-sortables-renders-simultaneously)

Here we can fix it by passing in an extra `groupName` argument to all the `sortable-group` and `sortable-items` helpers. So e.g.

```hbs
<ul {{sortable-group groupName="notas" onChange=this.setNotas}}>
  {{each ... }}
    <li
      ...
      {{sortable-item
        groupName="notas"
      }}
```

And `announcements` for the announcements bits.

Commit:
KAS-4496: resolve comments

sergiofenoll committed:
test: comment out all usages of setFormallyOkOnItemWithIndex

This PRs (temporarily) removes this functionality, so we need to disable
its usage in tests.

sergiofenoll commented on app/helpers/sum.js:
We already include the `ember-math-helpers` dependency that implements a helper with the same name and function, this custom helper can be removed.

https://robbiethewagner.github.io/ember-math-helpers/docs/api/helpers/sum

sergiofenoll committed:
chore: replacing sum helper with add from dependency

ember-math-helpers already includes a sum helper which does the same
thing as the one implemented here. It's also just an alias of the add
helper (also defined in that package) which we already use in the
codebase. Replacing it here with add for consistency.

sergiofenoll committed:
Revert "test: comment out all usages of setFormallyOkOnItemWithIndex"

This reverts commit 8d35109d716255d329e0531f6318d56cae54eff2.

sergiofenoll committed:
chore: enable formally OK selector + CSS changes to make it look ok

sergiofenoll commented on app/components/agenda/agenda-overview/agenda-overview-item.hbs:
@brenner-company in deze commit heb ik een paar CSS changes gemaakt zodat de selector rechts staat en om ervoor te zorgen dat de aflijning van alle items nog in orde blijft.

- Voeg `auk-o-flex--vertical-center` klasse toe aan wrapper div in `AgendaOverviewItem` om alles op één lijn te centreren
- Bottom margin verwijderen uit `.vlc-agenda-items__subject` (centreren gebeurt nu via de flex property op de parent div)
- Voeg `.auk-u-flex-grow` helper klasse toe en gebruik deze om de selector wrapper zo breed mogelijk te maken

Laat maar weten als dat zo gemerged mag worden of als je liever hebt dat dit anders aangepakt wordt. 

![image](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/assets/22411874/3eee4d08-a8be-4a47-8373-ed4bc5fc4244)

brenner-company replied:
Ziet er top uit ✅

sergiofenoll committed:
test: no longer wait for patch calls after moving agendaitems

sergiofenoll committed:
test: fix test with changes to agendaitem reordering

ValenberghsSven committed:
Merge branch 'development' of github.com:kanselarij-vlaanderen/frontend-kaleidos into feature/KAS-4496-efficientie-volgorde-agendapunten-aanpassen-verbeteren

ValenberghsSven committed:
Merge branch 'development' of github.com:kanselarij-vlaanderen/frontend-kaleidos into feature/KAS-4496-efficientie-volgorde-agendapunten-aanpassen-verbeteren

ValenberghsSven reviewed (changes requested):
Going to review this PR and make some changes based on what i've found after a small test:
- changing the order of announcements by dragging patches immediately (notas looks ok)
- it looks possible to search and reorder with the filtered items only

possibly more, not mergable as-is

ValenberghsSven committed:
KAS-4496 PR comments, disable search while reordering, fix cancel needs to reset variables

ValenberghsSven committed:
KAS-4496 add cypress changes to reflect moved functionality

ValenberghsSven reviewed (approved):
Changed:
- Cancel reset local variables
- Search agendaitems not possible when changing order
- Changing order button not possible on typing anything in the search bar
- Removed some unnecessary args
- Announcement dragging no longer saves immediately
- Hide newsletter status while reordering
- Renamed changes alert close > confirm so it made more sense with the added cancel method
- Updated cypress test with moved/renamed selectors


Wait for tests before merging. Many tests have changed with the selector update.
